infra/wca_on_rails/production/rails.tf

```diff
@@ -0,0 +1,265 @@
+    },
+    {
+      name = "DATABASE_WRT_USER"
+      value = var.DATABASE_WRT_USER
+    },
+    {
+      name = "VAULT_ADDR"
+      value = var.VAULT_ADDR
+    },
+    {
+      name = "VAULT_APPLICATION"
+      value = var.VAULT_APPLICATION
+    },
+    {
+      name = "TASK_ROLE",
+      value = aws_iam_role.task_role.name
+    }
+  ]
+  pma_environment = [
+    {
+      name = "PMA_USER_CONFIG_BASE64"
+      value = base64encode(templatefile("../templates/config.user.inc.php.tftpl",
+        { rds_host: "worldcubeassociation-dot-org.comp2du1hpno.us-west-2.rds.amazonaws.com",
+          rds_replica_host: "readonly-worldcubeassociation-dot-org.comp2du1hpno.us-west-2.rds.amazonaws.com" }))
+    }
+  ]
+}
+
+data "aws_iam_policy_document" "task_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "task_execution_role" {
+  name               = "${var.name_prefix}-task-execution-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "task_execution_role_attachment" {
+  role       = aws_iam_role.task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "task_role" {
+  name               = "${var.name_prefix}-task-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume_role_policy.json
+}
+
+data "aws_iam_policy_document" "task_policy" {
+  statement {
+    actions = [
+      "ssmmessages:CreateControlChannel",
+      "ssmmessages:CreateDataChannel",
+      "ssmmessages:OpenControlChannel",
+      "ssmmessages:OpenDataChannel",
+    ]
+
+    resources = ["*"]
+  }
+  statement {
+      actions = [
+        "s3:*",
+      ]
+
+      resources = [aws_s3_bucket.avatars.arn,
+                "${aws_s3_bucket.avatars.arn}/*",
+                   aws_s3_bucket.storage-bucket.arn,
+                "${aws_s3_bucket.storage-bucket.arn}/*"]
+    }
+}
+
+resource "aws_iam_role_policy" "task_policy" {
+  role   = aws_iam_role.task_role.name
+  policy = data.aws_iam_policy_document.task_policy.json
+}
+
+resource "aws_ecs_task_definition" "this" {
+  family = var.name_prefix
+
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["EC2"]
+
+  # We configure the roles to allow `aws ecs execute-command` into a task,
+  # as in https://aws.amazon.com/blogs/containers/new-using-amazon-ecs-exec-access-your-containers-fargate-ec2
+  execution_role_arn = aws_iam_role.task_execution_role.arn
+  task_role_arn      = aws_iam_role.task_role.arn
+
+  cpu = "2048"
+  memory = "7861"
+
+  container_definitions = jsonencode([
+    {
+      name              = "rails-production"
+      image             = "${var.shared.ecr_repository.repository_url}:latest"
+      cpu    = 1536
+      memory = 7861
```

Comment:
values for m6i

Reply:
changed to 8192 and 30000 until exact RAM value can be determined with use (should be something like 31xxx)